specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2020-07-01-preview/examples/CreateExtension.json

```diff
@@ -0,0 +1,61 @@
+{
+  "parameters": {
+    "subscriptionId": "subId1",
+    "resourceGroupName": "rg1",
+    "clusterRp": "Microsoft.Kubernetes",
+    "clusterResourceName": "connectedClusters",
+    "extensionInstanceName": "ClusterMonitor",
+    "api-version": "2020-07-01-preview",
+    "clusterName": "clusterName1",
+    "extensionInstance": {
+      "properties": {
+        "extensionType": "azuremonitor-containers",
+        "autoUpgradeMinorVersion": true,
+        "releaseTrain": "Preview",
+        "scope": {
+          "cluster": {
+            "releaseNamespace": "kube-system"
+          }
+        },
+        "configurationSettings": {
+          "omsagent.secret.wsid": "a38cef99-5a89-52ed-b6db-22095c23664b",
+          "omsagent.env.clusterName": "clusterName1"
```

Comment:
clusterResourceId and region parameter should be set by the config rp and shouldnt be exposed to the user.

Reply:
These are optional parameters that may be set by the user